mirgecom/simutil.py

```diff
@@ -74,90 +86,63 @@ def inviscid_sim_timestep(discr, state, t, dt, cfl, eos,
-                   step=0, t=0, dt=0, cfl=1.0, nstatus=-1, nviz=-1, exittol=1e-16,
-                   constant_cfl=False, comm=None, viz_fields=None, overwrite=False,
-                   vis_timer=None):
-    """Check simulation health, status, viz dumps, and restart."""
-    do_viz = check_step(step=step, interval=nviz)
-    do_status = check_step(step=step, interval=nstatus)
-    if do_viz is False and do_status is False:
-        return 0
 
-    dependent_vars = eos.dependent_vars(cv)
+def check_naninf_local(discr, dd, field):
+    """Check for any NANs or Infs in the field."""
+    s = op.nodal_sum_loc(discr, dd, field)
+    return np.isnan(s) or (s == np.inf)
 
-    rank = 0
-    if comm is not None:
-        rank = comm.Get_rank()
 
-    maxerr = 0.0
-    if exact_soln is not None:
-        actx = cv.mass.array_context
-        nodes = thaw(actx, discr.nodes())
-        expected_state = exact_soln(x_vec=nodes, t=t, eos=eos)
-        exp_resid = cv - expected_state
-        err_norms = [discr.norm(v, np.inf) for v in exp_resid.join()]
-        maxerr = discr.norm(exp_resid.join(), np.inf)
-
-    if do_viz:
-        io_fields = [
-            ("cv", cv),
-            ("dv", dependent_vars)
-        ]
-        if exact_soln is not None:
-            exact_list = [
-                ("exact_soln", expected_state),
-            ]
-            io_fields.extend(exact_list)
-        if viz_fields is not None:
-            io_fields.extend(viz_fields)
-
-        from mirgecom.io import make_rank_fname, make_par_fname
-        rank_fn = make_rank_fname(basename=vizname, rank=rank, step=step, t=t)
-
-        from contextlib import nullcontext
-
-        if vis_timer:
-            ctm = vis_timer.start_sub_timer()
-        else:
-            ctm = nullcontext()
-
-        with ctm:
-            visualizer.write_parallel_vtk_file(comm, rank_fn, io_fields,
-                overwrite=overwrite, par_manifest_filename=make_par_fname(
-                    basename=vizname, step=step, t=t))
-
-    if do_status is True:
-        #        if constant_cfl is False:
-        #            current_cfl = get_inviscid_cfl(discr=discr, q=q,
-        #                                           eos=eos, dt=dt)
-        statusmesg = make_status_message(discr=discr, t=t, step=step, dt=dt,
-                                         cfl=cfl, dependent_vars=dependent_vars)
-        if exact_soln is not None:
-            statusmesg += (
-                "\n------- errors="
-                + ", ".join("%.3g" % en for en in err_norms))
-
-        if rank == 0:
-            logger.info(statusmesg)
-
-    if maxerr > exittol:
-        raise ExactSolutionMismatch(step, t=t, state=cv)
+def compare_fluid_solutions(discr, red_state, blue_state):
+    """Return inf norm of (*red_state* - *blue_state*) for each component."""
+    resid = red_state - blue_state
+    return [discr.norm(v, np.inf) for v in resid.join()]
```

Comment:
Is this intended to be global? I'm not sure we *have* a local norm, but we can easily make one.

Reply:
I assumed that `discr.norm` is collective. I did want a collective compare here.